### ISSUE TITLE
Lifecycle methods not being called on widgets when using Flutter desktop

### ISSUE THREAD

**Issue by LeftRuleMatters** [flutter/flutter#87910]
## Steps to Reproduce

1. Add `WidgetsBindingObserver` as a mixin and override the`didChangeAppLifecycleState` function.
3. `debugPrint("$state");` in the function from the last step.
4. Run the application; unfocus, refocus, minimize, maximize, and close application
5. Feel the pain as nothing is printed.

**Expected results:**
Expected `inactive` or `paused` to be printed when the application lost focus and `resumed` to be printed when the application regained focus.

**Actual results:**
I had a few other debug statements, but nothing related to the `AppLifecycleState` enumeration.

Output from running with `--verbose`
```
PS D:\lifecycle_example> flutter run --verbose
[  +70 ms] executing: [C:\Flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +221 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] 0f465e5b2a3ed2431321b490a614c3d15089854c
[        ] executing: [C:\Flutter/] git tag --points-at 0f465e5b2a3ed2431321b490a614c3d15089854c
[  +53 ms] Exit code 0 from: git tag --points-at 0f465e5b2a3ed2431321b490a614c3d15089854c
[        ] 2.5.0-5.0.pre
[  +10 ms] executing: [C:\Flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +34 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/dev
[        ] executing: [C:\Flutter/] git ls-remote --get-url origin
[  +29 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] git@github.com:flutter/flutter.git
[ +308 ms] Unable to locate an Android SDK.
[   +3 ms] executing: [C:\Flutter/] git rev-parse --abbrev-ref HEAD
[  +29 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] dev
[  +52 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +92 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +4 ms] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +68 ms] Multiple devices found:
[  +24 ms] Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1151]
[   +1 ms] Chrome (web)      • chrome  • web-javascript • Google Chrome 92.0.4515.131
[        ] Edge (web)        • edge    • web-javascript • Microsoft Edge 92.0.902.67
[   +1 ms] [1]: Windows (windows)
[        ] [2]: Chrome (chrome)
[        ] [3]: Edge (edge)
[   +1 ms] Please choose one (To quit, press "q/Q")
[        ] :
[+20022 ms] 1
[  +11 ms] Skipping pub get: version match.
[  +95 ms] Generating D:\lifecycle_example\android\app\src\main\java\io\flutter\plugins\GeneratedPluginRegistrant.java
[  +89 ms] Initializing file store
[  +10 ms] Skipping target: gen_localizations
[   +6 ms] Skipping target: gen_dart_plugin_registrant        
[        ] Skipping target: _composite
[   +1 ms] complete
[   +5 ms] Launching lib\main.dart on Windows in debug mode...
[   +5 ms] C:\Flutter\bin\cache\dart-sdk\bin\dart.exe --disable-dart-dev C:\Flutter\bin\cache\artifacts\engine\windows-x64\frontend_server.dart.snapshot --sdk-root C:\Flutter\bin\cache\artifacts\engine\common\flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names --experimental-emit-debug-metadata-DFLUTTER_WEB_AUTO_DETECT=true --output-dill C:\Users\sorak\AppData\Local\Temp\flutter_tools.c676d554\flutter_tool.304b8299\app.dill --packages D:\lifecycle_example\.dart_tool\package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-scheme
org-dartlang-root --initialize-from-dill build\c075001b96339384a97db4862b8ab8db.cache.dill.track.dill --enable-experiment=alternative-invalidation-strategy
[  +24 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[  +25 ms] Exit code 0 from: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe -format json -products * -utf8 -latest -version 16 -requires Microsoft.VisualStudio.Workload.NativeDesktop Microsoft.VisualStudio.Component.VC.Tools.x86.x64 Microsoft.VisualStudio.Component.VC.CMake.Project
[        ] [
             {
               "instanceId": "e0541c05",
               "installDate": "2021-08-06T13:27:34Z",
               "installationName": "VisualStudio/16.10.4+31515.178",
               "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community",
               "installationVersion": "16.10.31515.178",
               "productId": "Microsoft.VisualStudio.Product.Community",
               "productPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\Common7\\IDE\\devenv.exe",
               "state": 4294967295,
               "isComplete": true,
               "isLaunchable": true,
               "isPrerelease": false,
               "isRebootRequired": false,
               "displayName": "Visual Studio Community 2019",
               "description": "Powerful IDE, free for students, open-source contributors, and individuals",
               "channelId": "VisualStudio.16.Release",
               "channelUri": "https://aka.ms/vs/16/release/channel",
               "enginePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
               "releaseNotes": "https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-v16.10#16.10.4",
               "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=660909",
               "updateDate": "2021-08-06T13:27:34.037643Z",
               "catalog": {
                 "buildBranch": "d16.10",
                 "buildVersion": "16.10.31515.178",
                 "id": "VisualStudio/16.10.4+31515.178",
                 "localBuild": "build-lab",
                 "manifestName": "VisualStudio",
                 "manifestType": "installer",
                 "productDisplayVersion": "16.10.4",
                 "productLine": "Dev16",
                 "productLineVersion": "2019",
                 "productMilestone": "RTW",
                 "productMilestoneIsPreRelease": "False",
                 "productName": "Visual Studio",
                 "productPatchVersion": "4",
                 "productPreReleaseMilestoneSuffix": "1.0",
                 "productSemanticVersion": "16.10.4+31515.178",
                 "requiredEngineVersion": "2.10.2174.31177"
               },
               "properties": {
                 "campaignId": "1357189423.1619605503",
                 "channelManifestId": "VisualStudio.16.Release/16.10.4+31515.178",
                 "nickname": "",
                 "setupEngineFilePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\setup.exe"
               }
             }
           ]
[   +5 ms] Building Windows application...
[  +13 ms] <- compile package:lifecycle_example/main.dart
[   +1 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe -S D:\lifecycle_example\windows -B build\windows -G Visual Studio 16 2019
[  +37 ms] -- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19043.
[  +35 ms] -- Configuring done
[ +770 ms] -- Generating done
[  +27 ms] -- Build files have been written to: D:/lifecycle_example/build/windows
[   +6 ms] executing: C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe --build build\windows --config Debug --target INSTALL --verbose
[ +126 ms] Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET Framework
[        ] Copyright (C) Microsoft Corporation. All rights reserved.
[  +58 ms] Build started 8/8/2021 9:40:10 AM.
[ +234 ms] Project "D:\lifecycle_example\build\windows\INSTALL.vcxproj" on node 1 (default targets).
[        ] Project "D:\lifecycle_example\build\windows\INSTALL.vcxproj" (1) is building "D:\lifecycle_example\build\windows\ZERO_CHECK.vcxproj" (2) on node 1 (default targets).
[        ] InitializeBuildStatus:
[        ]   Creating "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[  +28 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +65 ms]   Checking Build System
[  +69 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate".
[        ] Done Building Project "D:\lifecycle_example\build\windows\ZERO_CHECK.vcxproj" (default targets).
[ +115 ms] Project "D:\lifecycle_example\build\windows\INSTALL.vcxproj" (1) is building "D:\lifecycle_example\build\windows\ALL_BUILD.vcxproj" (3) on node 1 (default targets).
[        ] Project "D:\lifecycle_example\build\windows\ALL_BUILD.vcxproj" (3) is building "D:\lifecycle_example\build\windows\flutter\flutter_wrapper_app.vcxproj" (4) on node 1 (default targets).
[        ] Project "D:\lifecycle_example\build\windows\flutter\flutter_wrapper_app.vcxproj" (4) is building "D:\lifecycle_example\build\windows\flutter\flutter_assemble.vcxproj" (5) on node 1 (default targets).
[        ] InitializeBuildStatus:
[        ]   Creating "x64\Debug\flutter_assemble\flutter_assemble.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +2 ms] CustomBuild:
[        ]   "The build of 'D:\lifecycle_example\build\windows\CMakeFiles\12c77b8ccabf2a4c64c61cfcea0af84a\flutter_assemble.rule' depends on 'D:\LIFECYCLE_EXAMPLE\WINDOWS\FLUTTER\EPHEMERAL\FLUTTER_WINDOWS.DLL' which is produced by the build of
'D:\lifecycle_example\build\windows\CMakeFiles\d4b0af6489be5be3ca35500a5e9c1e09\flutter_windows.dll.rule'. The items cannot be built in parallel."
[  +31 ms]   Generating D:/lifecycle_example/windows/flutter/ephemeral/flutter_windows.dll, D:/lifecycle_example/windows/flutter/ephemeral/flutter_export.h, D:/lifecycle_example/windows/flutter/ephemeral/flutter_windows.h, D:/lifecycle_example/windows/flutter/ephemeral/flutter_messenger.h,
D:/lifecycle_example/windows/flutter/ephemeral/flutter_plugin_registrar.h, D:/lifecycle_example/windows/flutter/ephemeral/flutter_texture_registrar.h, D:/lifecycle_example/windows/flutter/ephemeral/cpp_client_wrapper/core_implementations.cc,
D:/lifecycle_example/windows/flutter/ephemeral/cpp_client_wrapper/standard_codec.cc, D:/lifecycle_example/windows/flutter/ephemeral/cpp_client_wrapper/plugin_registrar.cc, D:/lifecycle_example/windows/flutter/ephemeral/cpp_client_wrapper/flutter_engine.cc,
D:/lifecycle_example/windows/flutter/ephemeral/cpp_client_wrapper/flutter_view_controller.cc, _phony_
[+1151 ms]   [  +66 ms] executing: [C:\Flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +214 ms]   [ +235 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ]   [        ] 0f465e5b2a3ed2431321b490a614c3d15089854c
[        ]   [        ] executing: [C:\Flutter/] git tag --points-at 0f465e5b2a3ed2431321b490a614c3d15089854c
[  +60 ms]   [  +61 ms] Exit code 0 from: git tag --points-at 0f465e5b2a3ed2431321b490a614c3d15089854c
[        ]   [        ] 2.5.0-5.0.pre
[   +8 ms]   [   +9 ms] executing: [C:\Flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +42 ms]   [  +41 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ]   [        ] origin/dev
[        ]   [        ] executing: [C:\Flutter/] git ls-remote --get-url origin
[  +38 ms]   [  +39 ms] Exit code 0 from: git ls-remote --get-url origin
[        ]   [        ] git@github.com:flutter/flutter.git
[ +360 ms]   [ +361 ms] Unable to locate an Android SDK.
[   +3 ms]   [   +3 ms] executing: [C:\Flutter/] git rev-parse --abbrev-ref HEAD
[  +40 ms]   [  +40 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ]   [        ] dev
[  +52 ms]   [  +52 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +2 ms]   [   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +100 ms]   [ +102 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[   +1 ms]   [   +3 ms] Artifact Instance of 'WindowsUwpEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ]   [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +33 ms]   [  +36 ms] Initializing file store
[  +12 ms]   [  +12 ms] Done initializing file store
[  +41 ms]   [  +41 ms] Skipping target: gen_localizations
[  +10 ms]   [  +10 ms] Skipping target: gen_dart_plugin_registrant
[+2330 ms]   [+2330 ms] Skipping target: unpack_windows
[ +438 ms]   [ +438 ms] Skipping target: kernel_snapshot
[ +230 ms]   [ +230 ms] Skipping target: debug_bundle_windows_assets
[        ]   [        ] Persisting file store
[  +10 ms]   [  +10 ms] Done persisting file store
[   +3 ms]   [   +3 ms] build succeeded.
[   +8 ms]   [   +8 ms] "flutter assemble" took 3,250ms.
[ +251 ms]   [ +251 ms] ensureAnalyticsSent: 247ms
[   +5 ms]   [   +5 ms] Running shutdown hooks
[        ]   [        ] Shutdown hooks complete
[        ]   [   +1 ms] exiting with code 0
[  +65 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\flutter_assemble\flutter_assemble.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\flutter_assemble\flutter_assemble.tlog\flutter_assemble.lastbuildstate".
[        ] Done Building Project "D:\lifecycle_example\build\windows\flutter\flutter_assemble.vcxproj" (default targets).
[   +5 ms] InitializeBuildStatus:
[        ]   Creating "flutter_wrapper_app.dir\Debug\flutter_.91D7CF00.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +2 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +41 ms] ClCompile:
[        ]   All outputs are up-to-date.
[  +20 ms] Lib:
[        ]   All outputs are up-to-date.
[   +1 ms]   flutter_wrapper_app.vcxproj -> D:\lifecycle_example\build\windows\flutter\Debug\flutter_wrapper_app.lib
[  +10 ms] FinalizeBuildStatus:
[        ]   Deleting file "flutter_wrapper_app.dir\Debug\flutter_.91D7CF00.tlog\unsuccessfulbuild".
[        ]   Touching "flutter_wrapper_app.dir\Debug\flutter_.91D7CF00.tlog\flutter_wrapper_app.lastbuildstate".
[        ] Done Building Project "D:\lifecycle_example\build\windows\flutter\flutter_wrapper_app.vcxproj" (default targets).
[  +33 ms] Project "D:\lifecycle_example\build\windows\ALL_BUILD.vcxproj" (3) is building "D:\lifecycle_example\build\windows\flutter\flutter_wrapper_plugin.vcxproj" (6) on node 1 (default targets).
[        ] InitializeBuildStatus:
[        ]   Creating "flutter_wrapper_plugin.dir\Debug\flutter_.DD48F9D0.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +1 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[   +7 ms] ClCompile:
[        ]   All outputs are up-to-date.
[   +3 ms] Lib:
[        ]   All outputs are up-to-date.
[        ]   flutter_wrapper_plugin.vcxproj -> D:\lifecycle_example\build\windows\flutter\Debug\flutter_wrapper_plugin.lib
[   +5 ms] FinalizeBuildStatus:
[        ]   Deleting file "flutter_wrapper_plugin.dir\Debug\flutter_.DD48F9D0.tlog\unsuccessfulbuild".
[        ]   Touching "flutter_wrapper_plugin.dir\Debug\flutter_.DD48F9D0.tlog\flutter_wrapper_plugin.lastbuildstate".
[        ] Done Building Project "D:\lifecycle_example\build\windows\flutter\flutter_wrapper_plugin.vcxproj" (default targets).
[  +29 ms] Project "D:\lifecycle_example\build\windows\ALL_BUILD.vcxproj" (3) is building "D:\lifecycle_example\build\windows\runner\lifecycle_example.vcxproj" (7) on node 1 (default targets).
[        ] InitializeBuildStatus:
[        ]   Creating "lifecycle_example.dir\Debug\lifecycl.28A799B7.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +4 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +18 ms] ClCompile:
[        ]   C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30037\bin\HostX64\x64\CL.exe /c /ID:\lifecycle_example\windows /ID:\lifecycle_example\windows\flutter\ephemeral /ID:\lifecycle_example\windows\flutter\ephemeral\cpp_client_wrapper\include /Zi /nologo /W4 /WX
/diagnostics:column /Od /Ob0 /D WIN32 /D _WINDOWS /D _HAS_EXCEPTIONS=0 /D _DEBUG /D NOMINMAX /D UNICODE /D _UNICODE /D "CMAKE_INTDIR=\"Debug\"" /D _UNICODE /D UNICODE /Gm- /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++17 /Fo"lifecycle_example.dir\Debug\\"
/Fd"lifecycle_example.dir\Debug\vc142.pdb" /external:W4 /Gd /TP /wd4100 /errorReport:queue D:\lifecycle_example\windows\flutter\generated_plugin_registrant.cc D:\lifecycle_example\windows\runner\flutter_window.cpp
[  +64 ms]   generated_plugin_registrant.cc
[ +379 ms]   flutter_window.cpp
[ +719 ms]   Generating Code...
[  +76 ms] ResourceCompile:
[        ]   All outputs are up-to-date.
[  +25 ms] Link:
[        ]   C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30037\bin\HostX64\x64\link.exe /ERRORREPORT:QUEUE /OUT:"D:\lifecycle_example\build\windows\runner\Debug\lifecycle_example.exe" /INCREMENTAL /ILK:"lifecycle_example.dir\Debug\lifecycle_example.ilk" /NOLOGO
..\flutter\Debug\flutter_wrapper_app.lib D:\lifecycle_example\windows\flutter\ephemeral\flutter_windows.dll.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed
/manifestinput:D:\lifecycle_example\windows\runner\runner.exe.manifest /DEBUG /PDB:"D:/lifecycle_example/build/windows/runner/Debug/lifecycle_example.pdb" /SUBSYSTEM:WINDOWS /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"D:/lifecycle_example/build/windows/runner/Debug/lifecycle_example.lib" /MACHINE:X64  /machine:x64       
lifecycle_example.dir\Debug\Runner.res
[   +2 ms]   lifecycle_example.dir\Debug\flutter_window.obj
[        ]   lifecycle_example.dir\Debug\main.obj
[        ]   lifecycle_example.dir\Debug\utils.obj
[        ]   lifecycle_example.dir\Debug\win32_window.obj
[        ]   lifecycle_example.dir\Debug\generated_plugin_registrant.obj
[ +200 ms]   lifecycle_example.vcxproj -> D:\lifecycle_example\build\windows\runner\Debug\lifecycle_example.exe
[   +8 ms] FinalizeBuildStatus:
[        ]   Deleting file "lifecycle_example.dir\Debug\lifecycl.28A799B7.tlog\unsuccessfulbuild".
[        ]   Touching "lifecycle_example.dir\Debug\lifecycl.28A799B7.tlog\lifecycle_example.lastbuildstate".
[        ] Done Building Project "D:\lifecycle_example\build\windows\runner\lifecycle_example.vcxproj" (default targets).
[   +7 ms] InitializeBuildStatus:
[        ]   Creating "x64\Debug\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[   +3 ms] CustomBuild:
[        ]   All outputs are up-to-date.
[  +29 ms]   Building Custom Rule D:/lifecycle_example/windows/CMakeLists.txt
[  +35 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\ALL_BUILD\ALL_BUILD.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\ALL_BUILD\ALL_BUILD.tlog\ALL_BUILD.lastbuildstate".
[        ] Done Building Project "D:\lifecycle_example\build\windows\ALL_BUILD.vcxproj" (default targets).
[   +5 ms] InitializeBuildStatus:
[        ]   Creating "x64\Debug\INSTALL\INSTALL.tlog\unsuccessfulbuild" because "AlwaysCreate" was specified.
[  +47 ms] PostBuildEvent:
[        ]   setlocal
[        ]   "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DBUILD_TYPE=Debug -P cmake_install.cmake
[        ]   if %errorlevel% neq 0 goto :cmEnd
[        ]   :cmEnd
[        ]   endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone
[        ]   :cmErrorLevel
[        ]   exit /b %1
[        ]   :cmDone
[        ]   if %errorlevel% neq 0 goto :VCEnd
[        ]   :VCEnd
[  +41 ms]   -- Install configuration: "Debug"
[  +19 ms]   -- Up-to-date: D:/lifecycle_example/build/windows/runner/Debug/data/icudtl.dat
[        ]   -- Up-to-date: D:/lifecycle_example/build/windows/runner/Debug/flutter_windows.dll
[   +6 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets
[   +1 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/AssetManifest.json
[   +1 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/FontManifest.json
[   +1 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/fonts
[   +1 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/fonts/MaterialIcons-Regular.otf
[   +4 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/kernel_blob.bin
[  +73 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/NOTICES.Z
[   +1 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/packages
[        ]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/packages/cupertino_icons
[   +1 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/packages/cupertino_icons/assets
[   +1 ms]   -- Installing: D:/lifecycle_example/build/windows/runner/Debug/data/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf
[  +12 ms] FinalizeBuildStatus:
[        ]   Deleting file "x64\Debug\INSTALL\INSTALL.tlog\unsuccessfulbuild".
[        ]   Touching "x64\Debug\INSTALL\INSTALL.tlog\INSTALL.lastbuildstate".
[        ] Done Building Project "D:\lifecycle_example\build\windows\INSTALL.vcxproj" (default targets).
[   +1 ms] Build succeeded.
[        ]     0 Warning(s)
[        ]     0 Error(s)
[        ] Time Elapsed 00:00:08.07
[  +27 ms] Building Windows application... (completed in 9.2s)
[ +191 ms] Observatory URL on device: http://127.0.0.1:54509/-EYSnLcS0_U=/
[   +3 ms] Caching compiled dill
[  +43 ms] Connecting to service protocol: http://127.0.0.1:54509/-EYSnLcS0_U=/
[ +113 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:54509/-EYSnLcS0_U=/.
[  +62 ms] DDS is listening at http://127.0.0.1:54512/WU2dwAuk6FE=/.
[  +52 ms] Successfully connected to service protocol: http://127.0.0.1:54509/-EYSnLcS0_U=/
[  +15 ms] DevFS: Creating new filesystem on the device (null)
[  +16 ms] DevFS: Created new filesystem on the device (file:///C:/Users/sorak/AppData/Local/Temp/lifecycle_exampledac7c90b/lifecycle_example/)
[   +1 ms] Updating assets
[  +68 ms] Syncing files to device Windows...
[   +2 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +3 ms] <- recompile package:lifecycle_example/main.dart 207441a8-eb88-4edf-9008-c3f806b63d91
[        ] <- 207441a8-eb88-4edf-9008-c3f806b63d91
[   +1 ms] flutter: called initState
[  +54 ms] Updating files.
[        ] DevFS: Sync finished
[   +1 ms] Syncing files to device Windows... (completed in 63ms)
[        ] Synced 0.0MB.
[   +1 ms] <- accept
[   +2 ms] Connected to _flutterView/0x27149068ba0.
[   +2 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 
[        ] R Hot restart.
[        ] h List all available interactive commands.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ]  Running with sound null safety 
[        ] An Observatory debugger and profiler on Windows is available at: http://127.0.0.1:54512/WU2dwAuk6FE=/
[ +339 ms] DevTools activation throttled until 2021-08-08 21:05:48.731835.
[ +437 ms] The Flutter DevTools debugger and profiler on Windows is available at: http://127.0.0.1:9102?uri=http://127.0.0.1:54512/WU2dwAuk6FE=/
[+241805 ms] Skipping target: gen_localizations
[   +1 ms] Skipping target: gen_dart_plugin_registrant
[        ] Skipping target: _composite
[        ] complete
[   +1 ms] Performing hot restart...
[  +56 ms] Scanned through 521 files in 54ms
[        ] Syncing files to device Windows...
[        ] <- reset
[        ] Compiling dart to kernel with 0 updated files
[        ] <- recompile package:lifecycle_example/main.dart 4cd58b51-f777-425b-b1a8-f2cc66471184
[        ] <- 4cd58b51-f777-425b-b1a8-f2cc66471184
[  +77 ms] Updating files.
[  +20 ms] DevFS: Sync finished
[        ] Syncing files to device Windows... (completed in 100ms)
[        ] Synced 27.9MB.
[        ] <- accept
[ +120 ms] Hot restart performed in 277ms.
[   +2 ms] Performing hot restart... (completed in 281ms)
[        ] Restarted application in 288ms.
[ +322 ms] flutter: called initState
[+9997 ms] Service protocol connection closed.
[        ] Lost connection to device.
[   +2 ms] DevFS: Deleting filesystem on the device (file:///C:/Users/sorak/AppData/Local/Temp/lifecycle_exampledac7c90b/lifecycle_example/)
[ +263 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed
[   +5 ms] "flutter run" took 283,824ms.
[ +168 ms] ensureAnalyticsSent: 165ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 0
```

Output from running `flutter analyze`
```
PS D:\lifecycle_example> flutter analyze
Analyzing lifecycle_example...                                          
No issues found! (ran in 2.1s)
```

Flutter doctor output
```
PS D:\lifecycle_example> flutter doctor -v
[√] Flutter (Channel dev, 2.5.0-5.0.pre, on Microsoft Windows [Version 10.0.19043.1151], locale en-US)        
    • Flutter version 2.5.0-5.0.pre at C:\Flutter
    • Upstream repository git@github.com:flutter/flutter.git
    • Framework revision 0f465e5b2a (4 days ago), 2021-08-04 18:59:50 -0400
    • Engine revision a0d89b1a54
    • Dart version 2.14.0 (build 2.14.0-377.0.dev)

[X] Android toolchain - develop for Android devices
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.10.4)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.10.31515.178
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[√] IntelliJ IDEA Ultimate Edition (version 2021.2)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2021.1
    • Flutter plugin version 59.0.4
    • Dart plugin version 212.4746.57

[√] VS Code, 64-bit edition (version 1.60.0-insider)
    • VS Code at C:\Program Files\Microsoft VS Code Insiders
    • Flutter extension version 3.25.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1151]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 92.0.4515.131
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 92.0.902.67

! Doctor found issues in 2 categories.
```

Here is the minimum amount of code I used to produce this issue:
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> with WidgetsBindingObserver {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  void initState()
  {
    super.initState();
    WidgetsBinding.instance!.addObserver(this);
    debugPrint("called initState");
  }

  @override
  void deactivate()
  {
    debugPrint("called deactivate");
    super.deactivate();
  }

  @override
  void dispose()
  {
    debugPrint("called dispose");
    WidgetsBinding.instance!.removeObserver(this);
    super.dispose();
  }

  @override
  void didChangeAppLifecycleState(AppLifecycleState state)
  {
    debugPrint("AppLifecycleState: $state");
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}
```

**Comment by stuartmorgan** [comment]
Duplicate of https://github.com/flutter/flutter/issues/30735
